compiler/resolution/functionResolution.cpp

```diff
@@ -8913,10 +8919,18 @@ static void insertDynamicDispatchCalls() {
         BlockStmt* ifBlock = new BlockStmt();
         VarSymbol* cid = newTemp("_dynamic_dispatch_tmp_", dtBool);
         ifBlock->insertAtTail(new DefExpr(cid));
-        ifBlock->insertAtTail(new CallExpr(PRIM_MOVE, cid,
-                                new CallExpr(PRIM_TESTCID,
-                                             call->get(2)->copy(),
-                                             type->symbol)));
+
+        Expr* getCid = NULL;
+        if (isSuperAccess) {
+          // We're in a call on the super type.  We already know the answer to
+          // this if conditional
+          getCid = new SymExpr(gFalse);
```

Comment:
In this case I would prefer for the compiler not to create CondStmt at all, and instead just insert 'falseBlock' into the 'ifBlock'. I am OK with creating 'trueBlock' and not using it in this case, to minimize the need to add complexity.